### DISCUSSION TITLE
[Question]I want to DIY a remote controller that can use ELRS.（ i cant use Discard）

### DISCUSSION THREAD

**Discussion by zouzhe1** [ExpressLRS/ExpressLRS#2614] [Q&A] [answered]
## I can't use Discord, so I'm asking questions directly here. Sorry for the inconvenience.

1，ELRS only has open-source modules for receiving and transmitting, but not open-source remote controllers, right?
2，I want to make my own remote controller that can use ELRS modules. Could you recommend some materials or open-source addresses to me? Thank you.
3，The remote controller communicates with ELRS modules through a serial port. Is it the CSRF protocol?
Why can off-the-shelf commercial remote controllers use your ELRS modules?
4，Do we need to flash your firmware on those remote controllers?
5，So, have you written three pieces of firmware?
6，Are they for the receiving module, the transmitting module, and the transmitting remote controller?
7，If you can write the firmware for the transmitting remote controller, then do you have the schematics for those remote controllers?

I want to DIY a remote controller. please help me.
Thank you.

**Comment by pkendall64** [comment, marked as answer]
1. correct
2. There may be something on RC groups
3. Yes, it's CRSF protocol. For an internal module it's a full-duplex serial port, yes. External is a single wire half-duplex connection. Most controllers run EdgeTX and that supports the CRSF protocol.
4. No, it's flashed to the transmitter module that lives inside the radio or the external module.
5. We actually have 4. Transmitter module, Receiver, Transmitter module backpack, Goggles backpack.
6. We do not do the remote controller firmware. That is normally EdgeTX.
7. As above, we do not do this. It is EdgeTX. Not sure if there are open source controller projects that run EdgeTX.

**Reply by zouzhe1** [reply]
Thank you for your answer. I've figured out the relationships between these things now. I plan to reverse engineer the hardware pins based on the edgetx source code, draw the schematic diagram, and design the PCB.

**Reply by Hamo-creator** [reply]
Hello,
had you success with the reverse engineering of the hardware?
I am on a project like that and having some problems.
May you help me with them?
Thanks

**Comment by VergLsm** [comment]
@zouzhe1 请问你的PCB进展如何了？我也有这个想法

**Comment by pabloggargentina** [comment]
Me encanta la ide a de hacer un control remoto casero, en mi caso hago aeromodelismo desde hace unos 30 años y he modificado y adaptado algunos controles existentes a mis necesidades pero siempre con limitaciones a mis conocimientos o a la cantidad de elementos que habia disponible. hoy es mas facil acceder a tecnologia e internet permite comunicarnos con todo el planeta, respecto al tema creo que hoy habria que estimar la posibilidad de desarrollar un programa o app para android que funcione como un control remoto, creo que seria viable con el conocimiento y aporte y desarrollo de la gente indicada. creo que todos tenemos telefonos inteligentes o tabletas que han quedado en desuso en nuestros hogares, como asi tambien controles remotos usados de donde podriamos extraer joystys e interruptores, si ha eso le usamos una placa de desarrollo tipo arduino y un modulo elrs  como elemento de transmision quedaria un muy buen equipo de control remoto con una pantalla de buen tamaño con buenos colores y la posibilidad de ubicar la pantalla, los stick e interruptores a gusto y necesidad de cada constructor. la cuestion seria el desarrollo de ambos programas para el arduino y la app para el sistema android que permitan la interaccion.

**Comment by pabloggargentina** [comment]
I love the idea of ​​making a homemade remote control. In my case, I've been doing aeromodeling for about 30 years, and I've modified and adapted some existing controls to my needs, but always with limitations due to my knowledge or the number of elements available. Today, it's easier to access technology, and the internet allows us to communicate with the entire planet. Regarding this topic, I think we should consider the possibility of developing a program or app for Android that works as a remote control. I think it would be viable with the knowledge, contribution, and development of the right people. I think we all have smartphones or tablets that have fallen into disuse in our homes, as well as used remote controls from which we could extract joysticks and switches. If we use an Arduino-type development board and an ELRS module as a transmission element, we would have a very good remote control with a good-sized screen with good colors and the possibility of locating the screen, sticks, and switches to the taste and needs of each builder. The question would be the development of both programs for Arduino and the application for the Android system that allows interaction.
